mcs/class/corlib/System.Reflection/RuntimeMethodInfo.cs

```diff
@@ -698,7 +698,8 @@ public override MethodInfo MakeGenericMethod (Type [] methodInstantiation)
 #if FULL_AOT_RUNTIME
 				throw new NotSupportedException ("User types are not supported under full aot");
 #else
-				return new MethodOnTypeBuilderInst (this, methodInstantiation);
+				if (AppDomain.IsReflectionEmitSupported)
```

Comment:
Use existing RuntimeFeature.IsDynamicCodeSupported property

Reply:
Sure, didn't know that already existed.